cmd/meroxa/root/apps/deploy.go

```diff
@@ -615,36 +615,26 @@ func (d *Deploy) waitForDeployment(ctx context.Context, depUUID string) error {
 			var deployment *meroxa.Deployment
 			deployment, err := d.client.GetDeployment(ctx, d.appName, depUUID)
 			if err != nil {
-				return err
-			}
-			newLogs := strings.Split(deployment.Status.Details, "\n")
-			if len(newLogs) > len(logs) {
-				for i := len(logs); i < len(newLogs); i++ {
-					if len(logs) == 0 && i != 0 {
-						d.logger.StopSpinner(newLogs[i-1])
-					} else if len(logs) != 0 {
-						d.logger.StopSpinner(newLogs[i-1])
-					}
-					d.logger.StartSpinner("\t", newLogs[i])
-				}
-				logs = newLogs
+				return fmt.Errorf("couldn't fetch deployment status: %s", err.Error())
 			}
+			logs := strings.Split(deployment.Status.Details, "\n")
+
+			// Deployment was successful, we don't need to show API details
 			if deployment.Status.State == meroxa.DeploymentStateDeployed {
-				l := len(logs)
-				d.logger.StopSpinner(logs[l-1])
 				return nil
-			} else if deployment.Status.State == meroxa.DeploymentStateDeployingError ||
-				deployment.Status.State == meroxa.DeploymentStateRollingBackError {
-				l := len(logs)
-				d.logger.StopSpinnerWithStatus(logs[l-1], log.Failed)
-				return fmt.Errorf("failed to deploy Application %q", d.appName)
+			} else if deployment.Status.State == meroxa.DeploymentStateDeployingError {
```

Comment:
while we are here, why not make these switch stataments
```
switch deployment.Status.State {
case meroxa.DeploymentStateDeployingError: 
 // do work
... 
}
```
Since they are already exclusive.